src/server/job_func.c

```diff
@@ -2013,3 +2008,333 @@ find_aoe_from_request(resc_resv *presv)
 	return aoe_req;
 }
 #endif	/*ifndef PBS_MOM*/
+
+/**
+ * @brief	Check if the job is in the state specified
+ *
+ * @param[in]	pjob - pointer to the job
+ * @param[in]	state - the state to check
+ *
+ * @return	int
+ * @retval	1 if the job is in the state specified
+ * @retval	0 otherwise
+ */
+int
+check_job_state(const job *pjob, char state)
+{
+	if (pjob == NULL)
+		return 0;
+
+	if (get_job_state(pjob) == state)
+		return 1;
+
+	return 0;
+}
+
+/**
+ * @brief	Check if the job is in the substate specified
+ *
+ * @param[in]	pjob - pointer to the job
+ * @param[in]	substate - the substate to check
+ *
+ * @return	int
+ * @retval	1 if the job is in the state specified
+ * @retval	0 otherwise
+ */
+int
+check_job_substate(const job *pjob, int substate)
+{
+	if (pjob == NULL)
+		return 0;
+
+	if (get_job_substate(pjob) == substate)
+		return 1;
+
+	return 0;
+}
+
+/**
+ * @brief	Get the state character value of a job
+ *
+ * @param[in]	pjob - the job object
+ *
+ * @return char
+ * @return state character
+ * @return -1 for error
+ */
+char
+get_job_state(const job *pjob)
+{
+	if (pjob != NULL) {
+		return get_attr_c(&pjob->ji_wattr[JOB_ATR_state]);
+	}
+
+	return -1;
```

Comment:
No, -1 should be a valid signed char, right?

Reply:
it is, although It would be a non-printable char for the callers and I think callers don't check for the return value.
In any case, it's unlikely to happen.